backend/inspirehep/records/tasks.py

```diff
@@ -135,3 +142,17 @@ def regenerate_author_records_table_entries(uuids_to_regenerate):
         record.update_authors_records_table()
         record.update_self_citations()
         db.session.commit()
+
+
+def remove_duplicate_refs_from_record(matched_inspire_record, referenced_records_in_path, path):
+    affiliations_path = ".".join(path.split(".")[:-2])
+    affiliations = flatten_list(get_value(matched_inspire_record, affiliations_path, []))
+
+    if len(affiliations) == len(referenced_records_in_path):
+        return None
+
+    for affiliation in affiliations:
+        if flatten_list(affiliations).count(affiliation) > 1:
+            affiliations.remove(affiliation)
+    deep_set(matched_inspire_record, affiliations_path, affiliations)        
```

Comment:
I'm checking the api of `dict-deep` and I'm not sure if it accepts arrays, shouldn't we iterate through affiliations and paths and call `deep_set`? Maybe it is working in the tests because actually is only one in the list

Reply:
Hmm, I added an extra institution to the tests to see what happens it seems to work fine. I also added an assert to check that there are two institutions in the records references.